drivers/client_access_control/extensions/grda_warehouse/hud/client_extension.rb

```diff
@@ -45,7 +41,7 @@ module ClientExtension
         end
         # END_ACL
         limited_scope = limited_scope.where(id: client_ids) if client_ids.present?
-        merge(limited_scope)
+        current_scope.nil? ? limited_scope : where(id: limited_scope.select(:id))
```

Comment:
Question: should this be returning only source clients? If so, maybe we want to change this method a little

```diff
           if user.can_search_all_clients? || user.system_user? # rubocop:disable Style/IfInsideElse
-            current_scope || all
+            (current_scope || all).source
           else
```

Reply:
Given the comment in `drivers/client_access_control/app/controllers/client_access_control/clients_controller.rb` I _think_ this should return either source or destination.

```
  # Should always return any clients, source or destination that match
  def client_search_scope
    client_source.searchable_to(current_user)
  end
```

Additionally:
```
scope :destination_from_searchable_to, ->(user) do
        destination.where(id: searchable_to(user).
          joins(:warehouse_client_source).
          select(:destination_id))
      end
```
in `drivers/client_access_control/extensions/grda_warehouse/hud/client_extension.rb` would indicate it should return either.
Thanks for checking!

Reply:
Got it, thanks!